joku/cogs/games/stocks.py

```diff
@@ -176,6 +176,103 @@ def _identify_stock(self, channels: typing.Sequence[discord.abc.GuildChannel], n
+
+        If no target is provided, it will show your worth in assets
+        """
+        user = target or ctx.message.author
+        if user.bot:
+            await ctx.channel.send(":x: Bots cannot own shares.")
+            return
+
+        guild = await ctx.bot.database.get_or_create_guild(ctx.guild)
+        if not guild.stocks_enabled:
+            await ctx.send(":x: Stocks are not enabled for this server.")
+            return
+
+        # Display only Assets the user owns (stolen from currency)
+        if ctx.channel.permissions_for(ctx.guild.me).embed_links:
+            day = self.rng.randint(1, 31)
+            hour, minute, second = self.rng.randint(0, 23), self.rng.randint(0, 59), self.rng.randint(0, 59)
+
+            ts = datetime.datetime(year=2008, month=7, day=day, hour=hour, minute=minute, second=second)
+
+            em = discord.Embed(title="1st Stock Market of Joku")
+            em.description = "Your asset value is based on best estimates from current market prices."
+            em.set_author(name=user.display_name)
+
+            stocks = await ctx.bot.database.get_user_stocks(user)
+            em.add_field(name="Shares held", value=sum(userstock.amount for userstock in stocks))
+            em.add_field(name="Asset values", value="§{:.2f}".format(sum(userstock.amount * userstock.stock.price for userstock in stocks)))
+
+            em.set_thumbnail(url=user.avatar_url)
+            em.timestamp = ts
+            em.set_footer(text="Thanks capitalism!")
+            em.colour = user.colour
+            await ctx.send(embed=em)
+        else:
+            await ctx.channel.send(
+                  "User **{}** has `§{}` assets worth `§{}`.".format(user, 
+                      sum(userstock.amount for userstock in stocks), 
+                      sum(userstock.amount * userstock.stock.price for userstock in stocks)))
+
+    @assets.command(pass_context=True, aliases=["top", "leaderboard"])
+    async def richest(self, ctx: Context, *, what: str = "value"):
+        """
+        Shows the top 10 stock holders in this guild by number of assets held.
+        """
+        guild = await ctx.bot.database.get_or_create_guild(ctx.guild)
+        if not guild.stocks_enabled:
+            await ctx.send(":x: Stocks are not enabled for this server.")
+            return
+        
+        users = await ctx.bot.database.get_multiple_users(*ctx.message.guild.members)
+        
+        # Append a column to users holding asset total and asset worth
+        for user in users:        
+            stocks = await ctx.bot.database.get_user_stocks(user)
+            # Append a column amount to list users
+            user + [sum(userstock.amount for userstock in stocks]
+            # Append a column worth to list users
+            user + [sum(userstock.amount * userstock.stock.price for userstock in stocks)]
+                
+        if what == "amount" or "owned": #order by stock amount
```

Comment:
This doesn't evaluate to what you think it does:

```
>>> what == "amount" or "owned"
'owned'
```
This actually means `(what == "amount") or "owned"`, and since `"owned"` is a non-zero length string it will always evaluate to True if `what` isn't `"amount"`.